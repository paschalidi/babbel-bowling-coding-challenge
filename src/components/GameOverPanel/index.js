/**
 *
 * GameOverPanel
 *
 */
/* eslint-disable import/first */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */

import React from 'react';
import { Grid, Button } from 'semantic-ui-react';
import { connect } from 'react-redux';
import * as actions from '../../store/game-store/actions';
import * as R from 'ramda';


class GameOverPanel extends React.Component { // eslint-disable-line react/prefer-stateless-function
  getIndexOfWinner = () => {
    const array = Object.keys(this.props.players).map((index) => this.props.players[index].totalScore);
    return R.indexOf(Math.max.apply(Math, array), array);
  };

  render() {
    const winner = this.props.players[this.getIndexOfWinner()];
    return (
      <div className='flex panel background-color-black'>
        <Grid container centered stackable>
          <Grid.Row style={{ paddingBottom: '5vw' }}>
            <div className='glowing-text text-lg'>Game Over</div>
          </Grid.Row>

          <Grid.Row>
            <div className='glowing-text text-md'>The Winner Is {winner.name}</div>
          </Grid.Row>

          <Grid.Row style={{ paddingBottom: '5vw' }}>
            <div className='glowing-text text-md'>With Total Score {winner.totalScore}</div>
          </Grid.Row>


          <Grid.Row>
            <div className='glowing-text text-sm'>scores</div>
          </Grid.Row>
          <Grid.Row style={{ paddingBottom: '5vw' }} columns={this.props.amountOfPlayers}>
            {
              Object.keys(this.props.players).map(index =>
                <Grid.Column key={index} textAlign='center'>
                  <div className='glowing-text text-sm'>{this.props.players[index].name}
                    <br /> {this.props.players[index].totalScore}
                  </div>
                </Grid.Column>
              )
            }
          </Grid.Row>
          <Grid.Row>
            <Button
              inverted
              color='purple'
              onClick={() => this.props.resetGame()}
            >Start a New Game</Button>
          </Grid.Row>
        </Grid>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { ...state.game };
}

export default connect(mapStateToProps, actions)(GameOverPanel);
