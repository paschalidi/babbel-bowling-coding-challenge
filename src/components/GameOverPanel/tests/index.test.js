import React from 'react';
import renderer from 'react-test-renderer';
import GameOverPanel from '../index';
import { Provider } from 'react-redux';


import Store from '../../../store/configure';
const StoreInstance = Store();
test('GameOverPanel renders', () => {
  const component = renderer.create(
    <Provider store={StoreInstance}>
      <GameOverPanel />
    </Provider>
  );
  let tree;

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
