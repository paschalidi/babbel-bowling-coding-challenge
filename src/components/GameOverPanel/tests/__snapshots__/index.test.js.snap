// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOverPanel renders 1`] = `
<div
  className="flex panel background-color-black"
>
  <div
    className="ui centered container stackable grid"
  >
    <div
      className="row"
      style={
        Object {
          "paddingBottom": "5vw",
        }
      }
    >
      <div
        className="glowing-text text-lg"
      >
        Game Over
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="glowing-text text-md"
      >
        The Winner Is 
        Player 1
      </div>
    </div>
    <div
      className="row"
      style={
        Object {
          "paddingBottom": "5vw",
        }
      }
    >
      <div
        className="glowing-text text-md"
      >
        With Total Score 
        0
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="glowing-text text-sm"
      >
        scores
      </div>
    </div>
    <div
      className="row"
      style={
        Object {
          "paddingBottom": "5vw",
        }
      }
    >
      <div
        className="center aligned column"
      >
        <div
          className="glowing-text text-sm"
        >
          Player 1
          <br />
           
          0
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <button
        className="ui purple inverted button"
        disabled={undefined}
        onClick={[Function]}
        role="button"
        tabIndex={undefined}
      >
        Start a New Game
      </button>
    </div>
  </div>
</div>
`;
